server/container/core/src/main/java/org/apache/james/server/core/MimeMessageCopyOnWriteProxy.java

```diff
@@ -67,98 +92,208 @@ public MessageReferenceTracker(MimeMessage ref) {
             wrapped = ref;
         }
 
-        protected synchronized void incrementReferenceCount() {
-            referenceCount++;
+        protected void dispose() {
+            ReentrantReadWriteLock.WriteLock lock = this.lock.writeLock();
```

Comment:
I know this is code correct, but maybe I would rename the local variable `writeLock` to avoid any confusion with the class variable?